Studio-Lovelies/GG-JointJustice-Unity
# Adds SR demo to Case Select & automatically fetches Google Doc changes

## Summary
Adds two things:
- a button to start the Seductive Reasoning demo from the case select (82bf57f963415155227da2ec6009b12175163c08)  
- [`github/workflows/asset_sync.yml`](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/blob/82bf57f963415155227da2ec6009b12175163c08/.github/workflows/asset_sync.yml#L22-L23) that integrates the Google Doc of the SR Demo into Unity every 12 hours:
> 1. Stores a hash of the current .ink script
> 2. Runs [a Node.js script](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/blob/82bf57f963415155227da2ec6009b12175163c08/asset_sync/fetch_demo.js):
>    1. fetches the latest Google Doc revision
>    2. cleans up the formatting
>    3. writes it to the Asset folder
> 4. If the file hash hasn't changed: exits
> 5. Otherwise: Starts the editor with a script that instructs Unity to use Ink to parse the changes .ink into .json
> 6. Commits and pushes the files; that push to `develop` in turn kicks off the automatic release workflow

## Before/after screenshots and/or animated gif
### Before
![image](https://github.com/user-attachments/assets/8fae0c3b-87d9-4e69-bc72-85b67d71ccaf)

### After
![image](https://github.com/user-attachments/assets/e311b1c2-89d5-4a11-ad57-5e8b79a62102)

## Testing instructions
1. Open the `MainMenu` scene
4. Select `Case Select`
5. Click `SR Demo`
6. You will play previous Google Doc version of the Seductive Reasoning demo
7. After Github integrates this Pull Request, GitHub will create a release for this version, check for changes in the Google Doc (which it should find) and create another version with the latest script changes by @NightRoost

## Additional information

- `[ ]` Changes UI
- `[ ]` Introduces new feature
- `[ ]` Removes existing feature
- `[ ]` Has associated resource:
